File: customview/src/main/java/dinson/customview/utils/TypefaceUtils.java
package dinson.customview.utils;

import android.content.Context;
import android.graphics.Typeface;
import android.support.v4.util.SimpleArrayMap;

/**
 * @author Dinson - 2017/8/11
 */
public class TypefaceUtils {
    private static final SimpleArrayMap<String, Typeface> TYPEFACE_CACHE = new SimpleArrayMap<String, Typeface>();

    public static Typeface get(Context context, String name) {
        synchronized (TYPEFACE_CACHE) {
            if (!TYPEFACE_CACHE.containsKey(name)) {

                try {
                    Typeface t = Typeface.createFromAsset(context.getAssets(), name);
                    TYPEFACE_CACHE.put(name, t);
                } catch (Exception e) {
                    LogUtils.e("Could not get typeface '" + name + "' because " + e.getMessage());
                    return null;
                }
            }
            return TYPEFACE_CACHE.get(name);
        }
    }
}

File: blingbase/src/main/java/com/dinson/blingbase/utils/TypefaceUtils.kt
package com.dinson.blingbase.utils

import android.content.Context
import android.graphics.Typeface
import android.util.Log
import androidx.collection.SimpleArrayMap

/**
 * 字体工具（优化向）
 */
object TypefaceUtils {
    private val TYPEFACE_CACHE = SimpleArrayMap<String, Typeface>()

    private operator fun get(context: Context, name: String): Typeface? {
        synchronized(TYPEFACE_CACHE) {
            if (!TYPEFACE_CACHE.containsKey(name)) {
                try {
                    val t = Typeface.createFromAsset(context.assets, name)
                    TYPEFACE_CACHE.put(name, t)
                } catch (e: Exception) {
                    Log.e("TypefaceUtils", "Could not get typeface '" + name + "' because " + e.message)
                    return null
                }

            }
            return TYPEFACE_CACHE.get(name)
        }
    }
}
